qiskit_aer/primitives/sampler.py

```diff
@@ -99,26 +101,34 @@ def _call(
         if seed is not None:
             run_options.setdefault("seed_simulator", seed)
 
+        # Transpile circuits if not in cache
+        no_cache_circuits = []
+        for circuit_index in set(circuits):
+            if circuit_index not in self._cache:
+                no_cache_circuits.append(circuit_index)
+        if no_cache_circuits:
+            transpiled_circuits = (self._circuits[i] for i in no_cache_circuits)
+            if "shots" in run_options and run_options["shots"] is None:
+                transpiled_circuits = (
+                    self._preprocess_circuit(circ) for circ in transpiled_circuits
+                )
```

Comment:
I'm concerned that the runtime `run_options` here can affect the transpilation, which potentially invalidates the cache.

Reply:
exactly right